### PR TITLE
Removed duplicate check for min/max validators in rebalance

### PR DESCRIPTION
**Pull request by sampocs**
Closes: #419

## Context and purpose of the change
The min/max number of validators during rebalancing was checked twice - once in validate basic, and again in the actual message. There was also a different number used in each. 
Per the audit report, the fix was to remove the duplicate check (just keep it in validate basic) and add constants for the min/max values.

## Brief Changelog
* Added constants for the min/max number of validators to rebalance
* Removed duplicate check from msg_server
* Adjusted unit tests
* Added helper function to get admin address for validate basic unit tests

## Author's Checklist

I have...

- [x] Run and PASSED locally all GAIA integration tests
- [ ] If the change is contentful, I either:
    - [x] Added a new unit test OR 
    - [x] Added test cases to existing unit tests
- [ ] OR this change is a trivial rework / code cleanup without any test coverage

If skipped any of the tests above, explain.

## Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] manually tested (if applicable)
- [ ] confirmed the author wrote unit tests for new logic
- [ ] reviewed documentation exists and is accurate

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes? 
  - [ ] Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`?
  - [ ] This pull request updates existing proto field values (and require a backend and frontend migration)? 
  - [ ] Does this pull request change existing proto field names (and require a frontend migration)?
  How is the feature or change documented? 
      - [ ] not applicable
      - [ ] jira ticket `XXX` 
      - [ ] specification (`x/<module>/spec/`) 
      - [ ] README.md 
      - [ ] not documented

### CODE REVIEW

File: app/apptesting/test_helpers.go [hunk 1]
```diff
@@ -356,6 +357,14 @@ func GenerateTestAddrs() (string, string) {
 	return validAddr, invalidAddr
 }
 
+// Grabs an admin address to test validate basic on admin txs
+func GetAdminAddress() (address string, ok bool) {
+	for address := range utils.Admins {
+		return address, true
```
Comment (riley-stride) [inline app/apptesting/test_helpers.go:363]: nit: is this just going to always return the first admin address? kinda confusing to do that in a loop; possible it's a golang idiom

**Review by riley-stride — APPROVED** [review]
lgtm

File: app/apptesting/test_helpers.go (hunk 1 above)
Comment (sampocs) [reply app/apptesting/test_helpers.go:363]: the admins list is a map haha (not sure why tbh) but this is the only way to grab the "first item" from a map as far as I know

Comment (sampocs) [reply app/apptesting/test_helpers.go:363]: actually I guess is does make sense as a map cause we're generally starting from a supplied address and checking if it's an admin

**Pull request merged**